### PR TITLE
Automatically change IPs in SPT http.json to listen on all network interfaces.

### PR DESCRIPTION
**Pull request by robpneu**
I take no credit here; I just grabbed this from [OnniSaarni/SPT-Fika-Docker-Guide/blob/main/files/fcpy.sh#L21](https://github.com/OnniSaarni/SPT-Fika-Docker-Guide/blob/fec0307d1445e652144b1a72a642fe4b918ab668/files/fcpy.sh#L21.).

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -155,6 +156,11 @@ try_update_spt() {
     exit 0
 }
 
+spt_listen_on_all_networks() {
+    # Changes the ip and backendIp to 0.0.0.0 so that the server will listen on all network interfaces.
+    sed -i 's/127.0.0.1/0.0.0.0/g' $mounted_dir/SPT_Data/Server/configs/http.json
```
Comment (zhliau) [inline entrypoint.sh:161]: A good approach, but this assumes that the IP configured here is `127.0.0.1`, which would work for the default but may not if someone is moving over their SPT server install for example. How about using `jq` to be more explicit? e.g. 
```suggestion
# Haven't tested this, but something like...
json_path=$mounted_dir/SPT_Data/Server/configs/http.json
modified_http_json="$(jq '.ip = "0.0.0.0" | .backendIp = "0.0.0.0"' $json_path)" && echo -E "${modified_http_json}" > $json_path
```

Comment (robpneu) [reply entrypoint.sh:161]: You're right, it does assume that. It is always 127.0.0.1 in my case but may not be for everyone so I agree.

Comment (robpneu) [reply entrypoint.sh:161]: I like that solution. I did a bit of testing with it and it works. I'll add it to a commit

**Comment by robpneu** [comment]
> LGTM, thanks!

Sure thing! I don't have write access to this repo though so you'll have to merge it

**Pull request merged**